SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -199,15 +199,33 @@ void ActorSystem::ProcessAdds(const FEntitySubViewUpdate& SubViewUpdate)
 			{
 				RefreshEntity(Delta.EntityId);
 			}
+		}
+	}
+}
 
-			if (SubViewUpdate.SubViewType != ENetRole::ROLE_SimulatedProxy)
+void ActorSystem::ProcessAuthorityGains(const FEntitySubViewUpdate& SubViewUpdate)
+{
+	for (const EntityDelta& Delta : SubViewUpdate.EntityDeltas)
+	{
+		if ((Delta.Type == EntityDelta::ADD || Delta.Type == EntityDelta::TEMPORARILY_REMOVED)
+			&& SubViewUpdate.SubViewType != ENetRole::ROLE_SimulatedProxy)
+		{
+			const Worker_EntityId EntityId = Delta.EntityId;
+			if (SubViewUpdate.SubViewType == ENetRole::ROLE_Authority)
 			{
-				const Worker_ComponentSetId AuthorityComponentSet = SubViewUpdate.SubViewType == ENetRole::ROLE_Authority
-																		? SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID
-																		: SpatialConstants::CLIENT_AUTH_COMPONENT_SET_ID;
-
-				AuthorityGained(EntityId, AuthorityComponentSet);
+				// Check if this entity is EntitiesToRetireOnAuthorityGain first,
+				// to avoid authority gain on an actor that might've been deleted during a RepNotify.
+				if (HasEntityBeenRequestedForDelete(EntityId))
+				{
+					HandleEntityDeletedAuthority(EntityId);
```

Comment:
Was this previously not being called?

Reply:
it was (and I've still left it there) in ProcessAdds, but an actor could have been deleted during a repnotify (and hence its entityId would have ended up in the list that those functions use via its actorchannel closing)

Reply:
alternative is to just early out with a continue here, and wait till the next ProcessAdds does something (or maybe it would then just be in the removes entitydeltas)